From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 1 Dec 2022 16:52:27 +0100
Subject: [PATCH] Do not log offline mode warning

Ported from Winds-Studio/Leaf ver/26.2 (0036-Do-not-log-offline-mode-warning.patch), GPL-3.0-only.
Original author: Martijn Muijsers.
Canvas adaptation: gale.log.warning.offline.mode system property mapped to logs.disableOfflineModeWarning (default true).

diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 0bfd331fc8c434f29bb817cab072eb2b60339d20..8f642d4b2ae2e9c3099ed407ba4b7a489ffe9787 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -317,7 +317,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         String proxyFlavor = (io.papermc.paper.configuration.GlobalConfiguration.get().proxies.velocity.enabled) ? "Velocity" : "BungeeCord";
         String proxyLink = (io.papermc.paper.configuration.GlobalConfiguration.get().proxies.velocity.enabled) ? "https://docs.papermc.io/velocity/security" : "http://www.spigotmc.org/wiki/firewall-guide/";
         // Paper end - Add Velocity IP Forwarding Support
-        if (!this.usesAuthentication()) {
+        if (!io.canvasmc.canvas.GlobalConfiguration.getInstance().logs.disableOfflineModeWarning && !this.usesAuthentication()) { // Canvas - KeYi - do not log offline mode warning
             LOGGER.warn("**** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!");
             LOGGER.warn("The server will make no attempt to authenticate usernames. Beware.");
             // Spigot start
